### DISCUSSION TITLE
Per-Sensor Polling Rate

### DISCUSSION THREAD

**Discussion by Ndrinta** [nathanmarlor/foxess_modbus#71] [Ideas]
Hello again,
have you thought about about splitting the sensors in different polling rate tiers?

Something like:
**Real-Time** (fastest sampling rate available) / **Medium importance** (30-60 seconds polling rate) / **Low importance** (>1min polling rate)

This to tackle 2 issues, first being to not oversample low importance slow-changing or "noisy" sensors, bloating the db, and secondly, reducing the traffic on the network (which is always good when possible).

I believe this is the single feature a miss of [StealthChesnut](https://github.com/StealthChesnut)'s code.

**Comment by canton7** [comment]
Some things to bear in mind:
* When we poll for the new value of a sensor, we only tell HA about it if it has acually changed. In practice, there tends to be about 5 changed sensors per poll, and most of those are the ones you care about (PV1, etc)
* Network traffic is really minimal. Like, 14 packets of ~60 bytes each, spread over 10 seconds. Just open up Wireshark, and you'll see that these are completely swamped by everything else that's going on

**Reply by nathanmarlor** [reply]
Also worth adding that we gain performance by reading sensors in bulk, so two real-time sensors may be either side of a low importance sensor and so it makes logical sense to read it anyway.

**Reply by Ndrinta** [reply]
Thanks guys, I did not think about the fact that this code is optimized and it's reading a bunch of addresses at once.
